### PR TITLE
[chore] extract reusable header component

### LINKED ISSUE #44: Header component content shifts off screen
**Describe the bug**
When the app is run on an android device, in the *logged in state*, the profile picture component of the header is slightly pushed off-screen 

**To Reproduce**
1. Run the app locally and scan the QR code using your android device. You can also scan the `Development app` QR code available on the `README.md` file of this repository.
2. On the home screen, press `LOG IN` followed by `GO TO TABS` button
3. You will notice the profile picture component/icon is slightly off screen, when on the `Home tab`

**Expected behavior**
The header component should be inside the screen with a uniform horizontal margin. See the design mockup below:   

Screenshot 2022-10-10 at 10-10-13 Mobile

**Screenshots**
Below is the screenshot showing the error   

Screenshot 2022-10-10 at 10-17-41 WhatsApp Image 2022-10-10 at 10 14 48 jpeg (JPEG Image 606 × 1280 pixels) — Scaled (55%)

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
 - Device: [OPPO A12, model CPH2083]
 - OS: [Android 9, ColorOS version 6.1.2]

**Additional context**
N/A

### LINKED ISSUE #45: Extract Header Component into a Resuable Component
**Is your feature request related to a problem? Please describe.**
The header component is similar across different screens. This will result in repetition of code, violating the DRY principle

**Describe the solution you'd like**
Extract the code into a reusable component, such that any change or improvement will be done from one place

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by devKiratu**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

The reusable `MainHeader` component is extracted from similar header code across Screens, thus removing code repetition. 
This fixes #45 

It renders differently based on whether a user signed in or not. 
It is shared by the following screens:   
```
HomeScreenNotLoggedIn.tsx   
HomeScreen.tsx   
FeedScreen.tsx
AboutScreen.tsx
```

The `MainHeader` component also fixes #44 ( header components shifting off screen)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I have run the project locally and it works on both the Android emulator and an android device running on Android 9
- [ ] Test B

**Screenshots:**

**Home Screen**   

`Signed out`   

Screenshot 2022-10-13 at 13-14-38 WhatsApp Image 2022-10-13 at 12 46 19 (2) jpeg (JPEG Image 606 × 1280 pixels) — Scaled (55%)

`Signed in`    

Screenshot 2022-10-13 at 13-16-43 WhatsApp Image 2022-10-13 at 12 46 19 (1) jpeg (JPEG Image 606 × 1280 pixels) — Scaled (55%)

**Feed Screen**   

Screenshot 2022-10-13 at 13-17-22 WhatsApp Image 2022-10-13 at 12 46 18 (1) jpeg (JPEG Image 720 × 1520 pixels) — Scaled (46%)   

**About Screen**   

Screenshot 2022-10-13 at 13-18-16 WhatsApp Image 2022-10-13 at 12 46 17 jpeg (JPEG Image 606 × 1280 pixels) — Scaled (55%)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by devKiratu** [comment]
@brianwachira please review

File: src/components/layouts/MainHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+const MainHeader = (props: MainHeaderProps) => {
+	const { user } = useAppSelector((state) => state.user);
+
+	const navigation =
+		useNavigation<NativeStackNavigationProp<RootStackParamList>>();
+
+	//Temporary/experimental navigation to sign-in/sign-out screen from the profile picture icon
+	const goTOLoginScreen = () => navigation.navigate(screen_names.LOGIN);
```
Comment (makunomark) [inline src/components/layouts/MainHeader.tsx]: **Nit**: did you intend to name this as `goToLoginScreen`?

**Review by makunomark — APPROVED** [review]
LGTM

File: src/components/layouts/MainHeader.tsx (hunk 1 above)
Comment (devKiratu) [reply src/components/layouts/MainHeader.tsx]: Yes! Let me correct the typo

**Review by brianwachira — APPROVED** [review]
Good work @devKiratu

**Comment by devKiratu** [comment]
> Good work @devKiratu

Thanks!

**Pull request merged**